### PR TITLE
Fixed TypeLoadException 

### PR DESCRIPTION
**Pull request by devdaves**
This is my first pull request so I hope I get this right.  I ran into the same error as issue#43.  After some research I found this: https://aspnetwebstack.codeplex.com/workitem/1646 .  After a little more work I removed the WebApi.Selfhost package and replaced it with the WebApi.Owin.Selfhost package.  This got rid of the TypeLoadException and I was able to use the self host console app.

### CODE REVIEW

**Comment by devdaves** [comment]
Tried to run the tests after I made the pull request and the tests failed.  The test project did not have a reference to System.Net.Http.Primitives.  I added the Microsoft.Net.Http package to the test project to fix this.  Now the tests pass.  I didn't realize this commit would be part of the pull request.

File: src/WebApiBook.IssueTrackerApi.SelfHost/WebApiBook.IssueTrackerApi.SelfHost.csproj [hunk 1]
```diff
@@ -75,6 +106,23 @@
       <Name>WebApiBook.IssueTrackerApi</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <BootstrapperPackage Include=".NETFramework,Version=v4.5">
+      <Visible>False</Visible>
+      <ProductName>Microsoft .NET Framework 4.5 %28x86 and x64%29</ProductName>
+      <Install>true</Install>
+    </BootstrapperPackage>
+    <BootstrapperPackage Include="Microsoft.Net.Client.3.5">
```
Comment (glennblock) [inline src/WebApiBook.IssueTrackerApi.SelfHost/WebApiBook.IssueTrackerApi.SelfHost.csproj]: Why are these needed?

File: test/WebApiBook.IssueTrackerApi.AcceptanceTests/WebApiBook.IssueTrackerApi.AcceptanceTests.csproj [hunk 2]
```diff
@@ -95,7 +101,15 @@
       <Name>WebApiBook.IssueTrackerApi</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{82A7F48D-3B50-4B1E-B82E-3ADA8210C358}" />
```
Comment (glennblock) [inline test/WebApiBook.IssueTrackerApi.AcceptanceTests/WebApiBook.IssueTrackerApi.AcceptanceTests.csproj:105]: What's this?

Comment (devdaves) [inline test/WebApiBook.IssueTrackerApi.AcceptanceTests/WebApiBook.IssueTrackerApi.AcceptanceTests.csproj:105]: This was added by Visual Studio since I used Visual Studio's text explorer to run the tests after I added an xunit test extension.  Basically it defines the project as a unit test project which in turn will show specific context menu's in Visual Studio, more information is in the Microsoft answer to this persons question here: https://connect.microsoft.com/VisualStudio/feedback/details/800245/vs2013rc-adds-to-vs2012-c-project-section-itemgroup

It can be removed although anyone using the VS test explorer it will get added again.  If you would like me to remove it please let me know.

File: src/WebApiBook.IssueTrackerApi.SelfHost/WebApiBook.IssueTrackerApi.SelfHost.csproj (hunk 1 above)
Comment (devdaves) [inline src/WebApiBook.IssueTrackerApi.SelfHost/WebApiBook.IssueTrackerApi.SelfHost.csproj]: Removed these items.  Turned out my proj file wasn't saved prior to checking in.  Sorry about that.

**Comment by devdaves** [comment]
Just wondering what the status of this pull request is.

**Comment by glennblock** [comment]
Hi @devdaves sorry for the delay.I will take a look in the next few days.

File: src/WebApiBook.IssueTrackerApi.SelfHost/App.config [hunk 3]
```diff
@@ -5,7 +5,7 @@
     <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
       <dependentAssembly>
         <assemblyIdentity name="System.Net.Http.Formatting" publicKeyToken="31bf3856ad364e35" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-5.2.0.0" newVersion="5.2.0.0" />
+        <bindingRedirect oldVersion="0.0.0.0-4.0.0.0" newVersion="4.0.0.0" />
```
Comment (glennblock) [inline src/WebApiBook.IssueTrackerApi.SelfHost/App.config:8]: Why this change to the redirects?

**Comment by glennblock** [comment]
I am concerned with this change as the source for this sample should align as close as possible to the sample in the book. If we change the hosting to OWIN, then that breaks with the hosting approach in the chapter / which makes it less usable to the reader.

I'll think about it....

**Comment by glennblock** [comment]
@devdaves when are you seeing this error? When trying to just use the WebApiBook.IssueTrackerApi.SelfHost?

**Comment by devdaves** [comment]
@glennblock Yes.  If you set WebApiBook.IssueTrackerApi.SelfHost as the start up project and try to run it in debug mode you get the TypeLoadException.

**Comment by glennblock** [comment]
@devdaves can you try pulling the branch and getting your code back in sync? It looks like there is a merge conflict. I know it's a long time, but I'd like to get this merged.

**Comment by glennblock** [comment]
It looks like the TypeLoadException is fixed with PR #50. This didn't require pulling in the OWIN bits, but uses the latest self-host.

Can you see if that fixes your issue?
